.gitlab/ci/on-push.yml

```diff
@@ -1,8 +1,14 @@
+# This rule is to not run the build for docker update branches (for non-nightly packs)
+.filter-non-nightly-docker-updates-rule:
+  rules:
+    - if: '$CI_COMMIT_BRANCH =~ /^demisto\// && $CI_COMMIT_BRANCH !~ /^demisto\/.*-nightly$/'
+      when: never
+
 .push-rule:
   rules:
+    - !reference [.filter-non-nightly-docker-updates-rule, rules]
```

Comment:
What is the `rules` mentioned here?

Reply:
This is just the format the reference works. 
You Make a reference to the step, and what you are referencing to. 
In my case, I reference the rules I put in the step `.filter-non-nightly-docker-updates-rule`